### DISCUSSION TITLE
Plugin help

### DISCUSSION THREAD

**Discussion by Ramses800** [zim-desktop-wiki/zim-desktop-wiki#2537] [Q&A]
Hi, I wanted to write a simple plugin for converting selected text into a Code Block, but soon stumbled.
Just to change a selected text was easy, but getting it to be a code block made me befuddled.
```py
	@action(_('_Convert To Code'), accelerator='<Primary>#', menuhints='edit')  # T: Menu item
	def convert_to_code(self):
		'''Menu action to convert current selection to code fences'''
		buffer = self.pageview.textview.get_buffer()
		start, end = self._get_iters_one_or_more_lines(buffer)
		orig = start.get_text(end)		
		new = "{{{code: lang=\"js\" linenumbers=\"True\"" + orig + "}}}"
		with buffer.user_action:
			buffer.delete(start, end)
			buffer.insert(start, new)
```
Anyone who can give some hint on how to create a SourceView from the SourceViewPlugin?

**Comment by introt** [comment]
If you want a code block without reloading the page, you need to import the necessary class from the `sourceview` plugin and initialize the object yourself - ie. put the attributes in a `dict` and call `model_from_data` with that dict and `orig`. As for how that object is inserted into the buffer, you'll need to read the docs - good luck!
